test/integration/puppetlabs/services/master/environment_classes_int_test.clj

```diff
@@ -458,3 +480,43 @@
                   (response->class-info-map
                    test-response-after-all-flush))
                "unexpected body for test response")))))))
+
+(deftest ^:integration
+         not-modified-returned-for-environment-class-info-request-with-gzip-tag
+  (testing (str "SERVER-1153 - when the webserver gzips the response "
+                "containing environment_classes etag, the next request "
+                "roundtripping that etag returns an HTTP 304 (Not Modified)")
+    (let [expected-etag "abcd1234"
+          body-length 200000
+          jruby-service (reify jruby/JRubyPuppetService
+                          (get-environment-class-info-tag [_ _]
+                            expected-etag))
+          app (->
+               (fn [_]
+                 (master-core/response-with-etag
+                  (apply str (repeat body-length "a"))
+                  expected-etag))
+               (master-core/wrap-with-etag-check jruby-service))]
+      (jetty9/with-test-webserver
+       app
+       port
+       (let [request-url (str "http://localhost:" port)
+             initial-response (http-client/get request-url {:as :text})
+             response-tag (response-etag initial-response)
```

Comment:
@cprice404 I'd thought about doing that initially and ultimately didn't since I wasn't sure if we wanted a test to start failing because of a new behavior that didn't "break" anything in our implementation.  That said, though, it's probably a reasonable thing to do.  And maybe if it did start failing, it would prompt us to look into Jetty's new implementation further to see if there's something else we'd need to cover now and/or be able to undo the hacky "--gzip" handling code in Puppet Server.  That seems like it could be valuable.

As one other aside here...

I also debated whether to write this test using a service bootstrapped via `with-app-with-config` rather than the relatively simpler `jetty9/with-test-webserver` to make the test a little more friendly for switching it to run on some non-Jetty based implementation of the `:WebserverService` protocol in the future.  If instead, though, we're acknowledging that this test only really exists to validate Jetty's Etag manipulation behavior - including adding an assertion that checks for "--gzip" in the Etag value - then I don't think it would be worth switching over to `with-app-with-config`.

@cprice404 - you good with just adding the extra "--gzip" Etag value assertion and sticking with `jetty9/with-test-server` for now?